### PR TITLE
Migrate github actions to shared workflows

### PR DESCRIPTION
**Pull request by goruha**
## what
* Migrate github actions to shared workflows

## why
* Reduce levels of nested workflows

## references
* DEV-400: Move shared workflows into `.github` repo with `shared-` prefix

### CODE REVIEW

File: migrate/templates/github-action/.github/workflows/branch.yml [hunk 1]
```diff
@@ -16,6 +21,6 @@ permissions:
   actions: write
 
 jobs:
-  perform:
-    uses: cloudposse/github-actions-workflows-github-action-composite/.github/workflows/main-branch.yml@main
+  terraform-module:
```
Comment (osterman) [inline migrate/templates/github-action/.github/workflows/branch.yml:24]: this says terraform module

File: migrate/templates/github-action/.github/workflows/branch.yml [hunk 2]
```diff
@@ -16,6 +21,6 @@ permissions:
...
+    uses: cloudposse/.github/.github/workflows/shared-github-action.yml@main
```
Comment (osterman) [inline migrate/templates/github-action/.github/workflows/branch.yml:25]: This seems overly generic of a name. I had to do a double take when I saw the filename was "shared-github-action" - it sounds like it's a github action

File: migrate/templates/github-action/.github/workflows/branch.yml (hunk 1 above)
Comment (goruha) [inline migrate/templates/github-action/.github/workflows/branch.yml:24]:
```suggestion
  github-action:
```

File: migrate/templates/github-action/.github/workflows/release.yml [hunk 3]
```diff
@@ -9,5 +9,5 @@ permissions:
 
 jobs:
   perform:
```
Comment (goruha) [inline migrate/templates/github-action/.github/workflows/release.yml:11]:
```suggestion
  github-action:
```

**Review by osterman — APPROVED** [review]

**Pull request merged**